src/GraphQL.Client/GraphQLHttpClientOptions.cs

```diff
@@ -67,5 +67,11 @@ public class GraphQLHttpClientOptions
         /// See https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md#gql_connection_init.
         /// </summary>
         public Func<GraphQLHttpClientOptions, object?> ConfigureWebSocketConnectionInitPayload { get; set; } = options => null;
+
+        /// <summary>
+        /// The default user agent request header.
+        /// Default to the GraphQL client assembly.
+        /// </summary>
+        public Func<GraphQLHttpClientOptions, ProductInfoHeaderValue> DefaultUserAgentRequestHeader { get; set; } = _ => new ProductInfoHeaderValue(typeof(GraphQLHttpClient).Assembly.GetName().Name, typeof(GraphQLHttpClient).Assembly.GetName().Version.ToString());
```

Comment:
> I've added the requested changes. :) Do you think the function wrapper is needed? I have not so much experience with them so I'm not entirely sure if the current implementation is correct.

I don't think a function delegate is appropriate here, as the user agent request header sounds like something static across a single application...

I'm in favour of a regular property...

```suggestion
        public ProductInfoHeaderValue DefaultUserAgentRequestHeader { get; set; } = new ProductInfoHeaderValue(typeof(GraphQLHttpClient).Assembly.GetName().Name, typeof(GraphQLHttpClient).Assembly.GetName().Version.ToString());
```